src/test/java/marquez/db/models/JobRunRowTest.java

```diff
@@ -94,6 +94,20 @@ public void testNewJobRow_nullJobVersionUuid() {
         .build();
   }
 
+  @Test(expected = NullPointerException.class)
```

Comment:
Maybe it's better to throw an IllegalStateException instead of an NPE?

Reply:
This is during object instantiation, meaning there's no _initial_ state. With a null check, we are enforcing a constraint (or precondition) on what a parameter value can or cannot be i.e. _it cannot be null_

If you look at the JavaDoc for **IllegalStateException** it's not what we are checking for:
> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.